src/pages/Staking/MyStakesTable.tsx

```diff
@@ -0,0 +1,417 @@
+import Button, { ButtonType } from '@src/components/Button';
+import Dropdown from '@src/components/Dropdown';
+import Streak from '@src/components/Streak';
+import TableView from '@src/components/TableView';
+import {
+  CancelButtonXIcon,
+  GearIcon,
+  InstantWithdrawalIcon,
+} from '@src/components/icons';
+import CancelWithdrawalModal from '@src/components/modals/CancelWithdrawalModal';
+import InstantWithdrawalModal from '@src/components/modals/InstantWithdrawalModal';
+import StakingModal from '@src/components/modals/StakingModal';
+import UnstakeAllModal from '@src/components/modals/UnstakeAllModal';
+import useGateways from '@src/hooks/useGateways';
+import { useGlobalState } from '@src/store';
+import { ColumnDef, createColumnHelper } from '@tanstack/react-table';
+import dayjs from 'dayjs';
+import { useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+
+interface ActiveStakesTableData {
+  owner: string;
+  delegatedStake: number;
+  gateway: AoGateway;
+  pendingWithdrawals: number;
+  streak: number;
+}
+
+interface PendingWithdrawalsTableData {
+  owner: string;
+  gateway: AoGateway;
+  withdrawal: AoVaultData;
+  withdrawalId: string;
+}
+
+type TableMode = 'activeStakes' | 'pendingWithdrawals';
+
+const columnHelper = createColumnHelper<ActiveStakesTableData>();
+const columnHelperWithdrawals =
+  createColumnHelper<PendingWithdrawalsTableData>();
+
+const MyStakesTable = () => {
+  const walletAddress = useGlobalState((state) => state.walletAddress);
+  const ticker = useGlobalState((state) => state.ticker);
+
+  const { isFetching, data: gateways } = useGateways();
+  const [activeStakes, setActiveStakes] =
+    useState<Array<ActiveStakesTableData>>();
+  const [pendingWithdrawals, setPendingWithdrawals] =
+    useState<Array<PendingWithdrawalsTableData>>();
+
+  const [tableMode, setTableMode] = useState<TableMode>('activeStakes');
+
+  const [showUnstakeAllModal, setShowUnstakeAllModal] = useState(false);
+  const [stakingModalWalletAddress, setStakingModalWalletAddress] =
+    useState<string>();
+  const [showQuickStake, setShowQuickStake] = useState(false);
+  const [confirmCancelWithdrawal, setConfirmCancelWithdrawal] = useState<{
+    gatewayAddress: string;
+    vaultId: string;
+  }>();
+
+  const [confirmInstantWithdrawal, setConfirmInstantWithdrawal] = useState<{
+    gatewayAddress: string;
+    gateway: AoGateway;
+    vault: AoVaultData;
+    vaultId: string;
+  }>();
+
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    const activeStakes: Array<ActiveStakesTableData> | undefined = isFetching
+      ? undefined
+      : !walletAddress || !gateways
+        ? []
+        : Object.keys(gateways).reduce((acc, key) => {
+            const gateway = gateways[key];
+           
+            // @ts-expect-error - delegates is currently available on the gateway object
+            const delegate:AoGatewayDelegate = gateway.delegates[walletAddress?.toString()];
```

Comment:
can remove comment with latest sdk v

Reply:
Can not; the new SDK removes delegates and adds an API for retrieving delegate data separately from the gateway. The delegates is still being returned as part of the gateway data structure, and getGatewayDelegates() needs a change in the alpha SDK. These comments will be used to identify places in the app where calls to get delegates will eventually be replaced.